File: Variable.py
class Variable:
    def __init__(self, specs, var_type: str, pointer: bool, name: str):
        self.specs: str = specs
        self.var_type: str = var_type
        self.pointer: bool = pointer
        self.name: str = name

    def __str__(self):
        ptr_str = "*" if self.pointer else ""
        return self.specs + self.var_type + " " + ptr_str + self.name

File: SSC-C/src/main/java/cz/mipit/sscc/ssc/compiler/data/Variable.java
package cz.mipit.sscc.ssc.compiler.data;

import java.util.Objects;

public abstract class Variable {
    protected final String name;
    protected final boolean pointer;

    protected Variable(String name, boolean pointer) {
        this.name = Objects.requireNonNull(name);
        this.pointer = pointer;
    }

    public String getName() {
        return name;
    }

    public boolean isPointer() {
        return pointer;
    }
}
